app/javascript/packages/address-search/hooks/use-usps-locations.ts

```diff
@@ -0,0 +1,108 @@
+import { useState, useRef, useEffect, useCallback } from 'react';
+import { request } from '@18f/identity-request';
+import { t } from '@18f/identity-i18n';
+import useSWR from 'swr/immutable';
+import type { Location, FormattedLocation, LocationQuery, PostOffice } from '../types';
+import { formatLocations, snakeCase, transformKeys } from '../utils';
+
+const requestUspsLocations = async ({
+  locationsURL,
+  address,
+}: {
+  locationsURL: string;
+  address: LocationQuery;
+}): Promise<FormattedLocation[]> => {
+  const response = await request<PostOffice[]>(locationsURL, {
+    method: 'post',
+    json: { address: transformKeys(address, snakeCase) },
+  });
+
+  return formatLocations(response);
+};
+
+function requestAddressCandidates({
+  unvalidatedAddressInput,
+  addressSearchURL,
+}: {
+  unvalidatedAddressInput: string;
+  addressSearchURL: string;
+}): Promise<Location[]> {
+  return request<Location[]>(addressSearchURL, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    json: { address: unvalidatedAddressInput },
+  });
+}
+
+export default function useUspsLocations({
+  locationsURL,
+  addressSearchURL,
+}: {
+  locationsURL: string;
+  addressSearchURL: string;
+}) {
```

Comment:
are we relying on TS to detect the return type form the object at the end? would it be worthwhile describing the return type here? or defining one?